Shared/Supporting Files/Views/CategoryGridView.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct CategoryGridView: View {
+    /// All samples that will be shown in the categories.
+    private let samples: [Sample]
+    
+    /// The different sample categories generated from the samples list.
+    private let categories: [String]
+    
+    /// The columns for the grid.
+    private let columns = [
+        GridItem(.flexible()),
+        GridItem(.flexible())
+    ]
+    
+    init(samples: [Sample]) {
+        self.samples = samples
+        
+        var categoriesSet = Set<String>()
+        for sample in samples {
+            categoriesSet.insert(sample.category)
+        }
+        categories = Array(categoriesSet).sorted()
```

Comment:
```suggestion
        let categoriesSet = Set(samples.map(\.category))
        categories = categoriesSet.sorted()
```
When I love the expressibility and conciseness of Swift 🤩